app/controllers/api/configuration_script_payloads_controller.rb

```diff
@@ -2,6 +2,14 @@ module Api
   class ConfigurationScriptPayloadsController < BaseController
     include Subcollections::Authentications
 
+    def api_resource_action_options
+      if @req.action == "read" && @req.collection_id && @req.subcollection.blank?
+        %w[include_encrypted_attributes]
+      else
+        super
+      end
+    end
+
```

Comment:
NOTE this is overly broad as it disables all filters, would prefer to selectively allow specific columns while still filtering out actual v2 key values.

Reply:
we throw away strings (that should be encrypted) but keep hashes?

```ruby
def normalize_encrypted(value)
  value.kind_of?(String) ? nil : value
end
```

Reply:
@agrare It will not work if credentials are a mapping string

```json
      "Credentials": {
        "api_user.$": "$.api_user",
        "api_password.$": "$.api_password",
        "vcenter_user.$": "$.vcenter_user",
        "vcenter_password.$": "$.vcenter_password"
      },
```

/via https://github.com/ManageIQ/workflows-examples/blob/master/provision-vm-service/provision-vm.asl#L9C1-L14C9

But if it is a hash, then it may work:

```json
{"api_password": {"credential_ref": "manageiq_api", "credential_field": "userid"}}
```